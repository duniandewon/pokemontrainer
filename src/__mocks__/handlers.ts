import { http, HttpResponse } from "msw";

export const handlers = [
  http.post("https://beta.pokeapi.co/graphql/v1beta", ({ request }) => {
    // const { limit, offset } = request.body;

    return HttpResponse.json({
      data: {
        pokemon_v2_berry: [
          {
            pokemon_v2_item: {
              id: 1,
              name: "bulbasaur-berry",
              pokemon_v2_itemsprites: {
                sprites: "https://example.com/bulbasaur-berry.png",
              },
            },
            pokemon_v2_berryfirmness: {
              name: "soft",
            },
          },
        ],
      },
    });
  }),
];
