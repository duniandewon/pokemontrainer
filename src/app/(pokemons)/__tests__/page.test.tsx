import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";

import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

import Page from "../page";

import "@/__mocks__/intersectionObserverMock";

jest.mock("next/navigation", () => ({
  useRouter: () => ({
    push: jest.fn(),
  }),
}));

jest.mock("../usePokemonsVM", () => ({
  usePokemonsVM: () => ({
    pokemons: [
      { id: 1, name: "Bulbasaur" },
      { id: 2, name: "Charmander" },
    ],
    hasNext: true,
    selectedPokemon: -1,
    onSearchPokemons: jest.fn(),
    onSelectPokemon: jest.fn(),
    choosePokemon: jest.fn(),
    fetchNextPage: jest.fn(),
  }),
}));

describe("Page", () => {
  let queryClient: QueryClient;

  beforeEach(() => {
    queryClient = new QueryClient();
  });

  it("renders properly", () => {
    render(
      <QueryClientProvider client={queryClient}>
        <Page />
      </QueryClientProvider>
    );
  });

  it("renders a list of pokemon", () => {
    render(
      <QueryClientProvider client={queryClient}>
        <Page />
      </QueryClientProvider>
    );

    const bulbasaur = screen.getByAltText("Bulbasaur");

    expect(bulbasaur).toBeInTheDocument();
  });
});
