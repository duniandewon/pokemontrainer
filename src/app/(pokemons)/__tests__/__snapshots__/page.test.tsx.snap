// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders properly 1`] = `[Function]`;
